assignments/array-methods.js

```diff
@@ -2,82 +2,508 @@
+    shirt_size: "S",
+    company_name: "Kayveo",
+    donation: 74
+  },
+  {
+    id: 40,
+    first_name: "Minna",
+    last_name: "Hymas",
+    email: "mhymas13@cornell.edu",
+    shirt_size: "XS",
+    company_name: "Vimbo",
+    donation: 101
+  },
+  {
+    id: 41,
+    first_name: "Jsandye",
+    last_name: "Frend",
+    email: "jfrend14@ca.gov",
+    shirt_size: "XS",
+    company_name: "Latz",
+    donation: 156
+  },
+  {
+    id: 42,
+    first_name: "Yevette",
+    last_name: "Hacket",
+    email: "yhacket15@wp.com",
+    shirt_size: "XL",
+    company_name: "Lazzy",
+    donation: 291
+  },
+  {
+    id: 43,
+    first_name: "Hank",
+    last_name: "Zebedee",
+    email: "hzebedee16@ezinearticles.com",
+    shirt_size: "L",
+    company_name: "Gigashots",
+    donation: 241
+  },
+  {
+    id: 44,
+    first_name: "Jodie",
+    last_name: "Stawell",
+    email: "jstawell17@yale.edu",
+    shirt_size: "S",
+    company_name: "Jaxspan",
+    donation: 262
+  },
+  {
+    id: 45,
+    first_name: "Falito",
+    last_name: "Karsh",
+    email: "fkarsh18@pcworld.com",
+    shirt_size: "S",
+    company_name: "Mycat",
+    donation: 239
+  },
+  {
+    id: 46,
+    first_name: "Reginauld",
+    last_name: "Purselowe",
+    email: "rpurselowe19@thetimes.co.uk",
+    shirt_size: "L",
+    company_name: "Jabbersphere",
+    donation: 11
+  },
+  {
+    id: 47,
+    first_name: "Vida",
+    last_name: "Tydd",
+    email: "vtydd1a@dropbox.com",
+    shirt_size: "S",
+    company_name: "Quaxo",
+    donation: 55
+  },
+  {
+    id: 48,
+    first_name: "Anderea",
+    last_name: "MacGiolla Pheadair",
+    email: "amacgiollapheadair1b@xing.com",
+    shirt_size: "2XL",
+    company_name: "Kwimbee",
+    donation: 214
+  },
+  {
+    id: 49,
+    first_name: "Bel",
+    last_name: "Alway",
+    email: "balway1c@ow.ly",
+    shirt_size: "S",
+    company_name: "Voolia",
+    donation: 107
+  },
+  {
+    id: 50,
+    first_name: "Shell",
+    last_name: "Baine",
+    email: "sbaine1d@intel.com",
+    shirt_size: "M",
+    company_name: "Gabtype",
+    donation: 171
+  }
+];
 
 // ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName.
 let fullName = [];
+runners.forEach(function(runner) {
+  fullName.push(`${runner.first_name} ${runner.last_name}`);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
+const firstName = runners.map(function(runner) {
```

Comment:
You don't need to create a new array. you can do something like

```javascript
const firstName = runners.map(function(runner) {
   return (toUpper(runner.first_name));
}
```

`The map() method creates a new array with the results of calling a function for every array element.

The map() method calls the provided function once for each element in an array, in order.`